src/vcf/validator.cpp

```diff
@@ -141,6 +143,9 @@ namespace ebi
         std::vector<char> line;
         ebi::util::readline(input, line);
         ebi::vcf::Version version;
+        if (ebi::vcf::is_compressed_file(sourceName, line)) {
+            return false;
```

Comment:
We talked about how this will be shown at the end to the user:
> [info] Reading from input file...
> [warning] Input file should not be compressed (detected .gz compression)
> [info] According to the VCF specification, the input file is not valid

And we are not sure the last line is very clear. It's not really true that the VCF is not valid, it's just in a format we can't accept right now. So another change is needed. Here, instead of returning false (which means it's not valid) just throw a `std::invalid_argument`. If you see, [the main function](https://github.com/EBIvariation/vcf-validator/blob/develop/src/validator_main.cpp#L243) is prepared so that it logs an error if an exception is thrown.

Also another couple of changes will be needed, in `compressed_file_warning` you can just say 
`BOOST_LOG_TRIVIAL(warning) << "detected " << file_extension << " compression";` so that it's not redundant, and change the tests so that they check that exceptions are thrown, instead of returning false. 

With this change, the output should be:
> [info] Reading from input file...
> [warning] detected .gz compression
> [error] Input file should not be compressed

Which is much more clear IMO.